zio-http/src/main/scala/zio/http/codec/internal/EncoderDecoder.scala

```diff
@@ -243,8 +335,40 @@ private[codec] object EncoderDecoder                   {
       } else if (jsonDecoders.length == 1) {
         jsonDecoders(0)(body).map { result => inputs(0) = result }
       } else {
-        ZIO.foreachDiscard(jsonDecoders.zipWithIndex) { case (decoder, index) =>
-          decoder(body).map { result => inputs(index) = result }
+        val indexByName = flattened.content.zipWithIndex.map { case (codec, idx) =>
+          codec.name.getOrElse("field" + idx.toString) -> idx
+        }.toMap
+        body.asMultipartFormStream.flatMap { form =>
+          form.fields.runForeach { field =>
+            indexByName.get(field.name) match {
```

Comment:
Would be nice to avoid the `Option` boxing here. Maybe `indexByName(0)` and you can catch the exception appropriately and turn it into `HttpCodecError.MalformedBody(s"Unexpected multipart/form-data field: ${field.name}")`.

Reply:
No way to fix this one?